kempner_computing_handbook/s1_high_performance_computing/development_and_runtime_envs/software_module_and_environment_management.md

```diff
@@ -16,14 +21,33 @@ See a list of software modules available on the FASRC Cannon cluster [here](http
 The following commands are used to manage software modules:
 
 
-| Command                       | Description                          | 
-|-------------------------------|--------------------------------------|
-| `module avail`                | List all available software modules. |
-| `module load <module_name>`   | Load a software module.              |
-| `module unload <module_name>` | Unload a software module.            |
-| `module list`                 |  List all loaded software modules.   |
-| `module purge`                | Unload all loaded software modules.  |
+| Command                       | Description                                                  |
+|-------------------------------|--------------------------------------------------------------|
+| `module avail`                | List all available software modules.                         |
+| `module avail <string>`       | List all available software modules that contain `<string>` in their name. |
+| `module load <module_name>`   | Load a software module.                                      |
+| `module unload <module_name>` | Unload a software module.                                    |
+| `module list`                 | List all loaded software modules.                            |
+| `module purge`                | Unload all loaded software modules.                          |
+
+
+```{tip}
+You can use `ml` as a shortcut for common commands:
+
+| Command                       | Shortcut            |
+|-------------------------------|---------------------|
+| `module list`                 | `ml`                |
+| `module load <module_name>`   | `ml <module_name>`  |
+| `module unload <module_name>` | `ml -<module_name>` |
+```
 
+## How do modules work?
 
+Each module is defined by a script that runs when the module is loaded.
+This script modifies environment variables such as `PATH`, `LD_LIBRARY_PATH`, etc. to make the software accessible from your terminal session.
+You can use `module show <module_name>` to show a module's script.
 
+These changes to the terminal environment will persist until `module unload` is called.
```

Comment:
Please also add `module purge`. Instead of listing one sentence per line, continue the text as a cohesive paragraph. If there’s a need to emphasize a specific sentence, you can format it as a separate item, but otherwise keep the flow continuous for better readability.